tokio/src/future/trace.rs

```diff
@@ -0,0 +1,11 @@
+use std::future::Future;
+
+pub(crate) trait InstrumentedFuture: Future {
+    fn id(&self) -> Option<tracing::Id>;
+}
+
+impl<F: Future> InstrumentedFuture for tracing::instrument::Instrumented<F> {
+    fn id(&self) -> Option<tracing::Id> {
+        self.span().id()
+    }
+}
```

Comment:
hmm, we should maybe just go ahead and add this to `tracing`'s `Instrumented` type...

doing it in tokio is absolutely fine, of course, but an upstream PR would also be welcome eventually! i wonder if we should just add `Deref<Span>` and `Deref<T>` impls for `Instrumented<T>`...but that might be `Deref` abuse...

Reply:
Add _what_ to `tracing::Instrumented`? You can already access the `span` on it.

I guess I could have changed all the functions in `runtime` to task `Instrumented<F>`, but that seems more complex with conditional compilation. Just having a trait seemed easier.

Reply:
ahh, i misinterpreted the intent behind this --- i thought the intention behind the trait was just to be an extension trait adding an `id()` method that you could call instead of `span().id()`, not to abstract over various types that may or may not be instrumented... never mind!